### ISSUE TITLE
app unavailable on French App store

### ISSUE THREAD

**Issue by krille-chan** [krille-chan/fluffychat#76]
Hi I'd be more than willing to try out the Mac app version, but it's unavailable at the French app store...
Will it be soon ?

**Comment by squid-f** [comment]
Hi. Not sure it could be related but I have the same need to be able to use FluffyChat on iOS from the French app store.
Thanks!

**Comment by nathanael-h** [comment]
Hello, I have dev accounts on both Apple and android stores. May I help about this?
Also I am a native French speaker, maybe this can help to fill specific papers about crypto, if this is the issue.

**Comment by nononymousse** [comment]
just looked it up (french appstore) but don't see it and when i go on the website and click on "get on appstore", i have the appstore pop-up but it stay empty (iOS 15.7.9, iPhone SE (2016))

**Comment by Mikaela** [comment]
I think this should be pinned and made excempt from stalebot rather than closed for inactivity unless there is a FAQ with this question.

**Comment by casimir** [comment]
There was another issue on gitlab that has been lost in the migration. I was titled "Android version not available for France in the Play Store" and I felt it was promising. I still have a few notification emails in my mailbox and I could find the most important comment from @mateMathieu (same person here 🤞), a good summary of the whole thread.

Here is the unedited comment:
> [mateMathieu](https://gitlab.com/mateMathieu) commented on a [discussion](https://gitlab.com/famedly/fluffychat/-/issues/220#note_1455446343):
> 
> OK, I just talked to him on the phone. He is a very nice person, mindful and affable.
> 
> He understand the reticence to publish in France because it is managed in the government by the ministry of Army (Secrétariat général de la défense et de la sécurité nationale). This is due to history of such legislation, because the people that used these encryption technologies were the army, the secret services and the diplomats. However, the law text are old (2005 and 2007), and quite not adapted to the new technology and the fast pace of evolution that is going on in computer science and cryptographic technologies of our time. As an example, 2007 was before Blockchain.
> 
> The only requirement they impose, is that you share the code with them if they ask you. Beyond this it is a simple declaration that they don't oppose or evaluate. So the legally binding document is simply a declarative document that the administration provide to the developers to list them as software using encryption technology to the french. Nothing else.
> 
> In essence what he said, as a lawyer and accustomed to this very problem, is that the developers of open-source projects have their code public, hence the authorities can pull at any time the code and get the information they want. What the developers are mandated to do, if their code is closed, is to produce the content of the public key that are used in the cryptographic algorithm used in the software. As long as there is a private key that is not known by the developer, in the case of asymmetric encryption, they are not legally bound to produce those keys, simply because they cannot be force to know these, and they don't have them, simply. Today, everything is asymmetric encryption, so the law is even more useless...
> 
> TLDR: I hope that the devs will look at this message with a fresh eye and realize that the outdated legal framework does not force or involve their person or their project in any limitation or constrain. It is just a declarative paperwork that might not even be read by the ministry in charge, that happen to be historically attached to the military part of the administration.
> 
> Here is the documentation and the webpage that explain this: https://www.ssi.gouv.fr/entreprise/reglementation/controle-reglementaire-sur-la-cryptographie/les-formulaires/
> 
> Let me know if I can assist furthermore in any way.
> 
> Cheers PS: I didn't want to share his name and business publicly, but I can to the devs if they want to.


Also as a french citizen, heavy matrix user and app developer myself, I can appreciate the bother of this kind of situation for OSS maintainers and I'd like to propose my help.
With @krille-chan authorization I could publish the app on french stores myself. I already have store accounts set up and paid for anyway. I'd serve as the publisher and handle the administrative compliance myself but take 0 credit for the app developpement itself.
This is just a proposition for an alternative. I'm not really eager to go this way but I remember the good old days when Fluffychat was available via TestFlight. And I'd rather use Fluffychat rather that Element as my daily driver.

**Comment by c4ffein** [comment]
Hi, I'd like to get this app on iPhone too, without having to sideload if possible...
- How is @krille-chan related to [Famedly GmbH](https://apps.apple.com/us/developer/famedly-gmbh/id1551469602)?
- Can we reach out to them, as french who could help them going through this legal process?
- If they really don't want to go through this (and I totally understand, who desn't hate our outdated french administration), I think it would be great to do this @casimir . I can also do it or just help if needed.

**Comment by krille-chan** [comment]
@c4ffein thank you so much. There is a matrix group about this topic already but it is very silent. If you are interested to help, I can invite you. Just need your matrix ID

**Comment by c4ffein** [comment]
Of course @krille-chan , same username as here, but on the matrix.org instance :)

**Comment by Follpvosten** [comment]
hey, what's the current status on this?

**Comment by c4ffein** [comment]
I invited you, tell me if that worked

**Comment by mateMathieu** [comment]
@c4ffein no it did not, can you try again?

**Comment by deadbaed** [comment]
Hello, is it useful to join the matrix room to follow progress on this? Or there is not a lot of progress and staying subscribed to this issue is fine?

**Comment by Reaster0** [comment]
hi, i'm another one coming here to see i'm not the only one

**Comment by krille-chan** [comment]
> hi, i'm another one coming here to see i'm not the only one

Can you translate and fill out the PDF forms for me and do the investigation whom I have to send it?

**Comment by mateMathieu** [comment]
The file and all the explanations are available here now: https://cyber.gouv.fr/reglementation/reglementation-identite-confiance-numerique/controles-reglementaires-cryptographie/controle-moyen-de-cryptologie/

The website is even less easy to understand than before... But I found a solution. I compiled and produced the document that people need to fill out and send to the administration.
So you can find attached the file as PDF or LibreOffice Writer document, translated in English as well to help (scroll down).

Let me know if you found this useful. Of course it only works with the current version of the rule. But the philosophy of the document is identical. You can find the content of the document in the first annex of the law text, in case you have to refresh this.

Mat

[DECLARATION ET DEMANDE D'AUTORISATION D'OPERATIONS RELATIVES A UN MOYEN DE CRYPTOLOGIE.pdf](https://github.com/user-attachments/files/25050395/DECLARATION.ET.DEMANDE.D.AUTORISATION.D.OPERATIONS.RELATIVES.A.UN.MOYEN.DE.CRYPTOLOGIE.pdf)

[DECLARATION ET DEMANDE D'AUTORISATION D'OPERATIONS RELATIVES A UN MOYEN DE CRYPTOLOGIE.odt](https://github.com/user-attachments/files/25050397/DECLARATION.ET.DEMANDE.D.AUTORISATION.D.OPERATIONS.RELATIVES.A.UN.MOYEN.DE.CRYPTOLOGIE.odt)

**Comment by TaciteOFF** [comment]
Hello! I'd like to bump this because I'd love to use FluffyChat and it's not available :(

**Comment by TaciteOFF** [comment]
Would it be possible to provide an official and up-to-date IPA file so we can at least sideload it?

**Comment by amurancsikHIP** [comment]
Bump! Checking today the app is still not available in french app stores.

**Comment by superlemon57** [comment]
Up

**Comment by midnya** [comment]
Bump-ing as well

What's the work that needs to be done? Can we contribute to this?

**Comment by TaciteOFF** [comment]
@krille-chan Would it be possible to provide an official and up-to-date IPA file so we can at least sideload it?
